ext/sodium/libsodium.c

```diff
@@ -3405,23 +3284,23 @@ PHP_FUNCTION(sodium_crypto_secretstream_xchacha20poly1305_push)
 	}
 	ZVAL_DEREF(state_zv);
 	if (Z_TYPE_P(state_zv) != IS_STRING) {
-		zend_throw_exception(sodium_exception_ce, "a reference to a state is required", 0);
+		zend_argument_error(sodium_exception_ce, 1, "must be a reference to a state");
 		RETURN_THROWS();
 	}
 	sodium_separate_string(state_zv);
 	state = (unsigned char *) Z_STRVAL(*state_zv);
 	state_len = Z_STRLEN(*state_zv);
 	if (state_len != sizeof (crypto_secretstream_xchacha20poly1305_state)) {
-		zend_throw_exception(sodium_exception_ce, "incorrect state length", 0);
+		zend_argument_error(sodium_exception_ce, 1, "to have a correct length");
```

Comment:
Probably this should use negative form now.

Reply:
I think "... must have a correct length" should do it as well.

Reply:
But if you have a better suggestion, I'm definitely open for a change.